examples/spdif_tx_example/src/main.xc

```diff
@@ -56,25 +62,34 @@ void generate_samples(chanend c) {
     }
 }
 
-void audio_tasks(client output_gpio_if gpio[2]) {
-   chan c;
-   gpio[0].output(1);
-   gpio[1].output(1);
-   configure_clock_src(clk_audio, p_mclk_in);
-   set_clock_fall_delay(clk_audio, 7);
-   start_clock(clk_audio);
-   par 
-   {
-      spdif_tx(p_spdif_tx, c);
-      generate_samples(c);
-   }
+void audio_tasks(client output_gpio_if i_gpio[8]) 
+{
+    chan c;
+
+    for (int i= 0; i < 8; i++)
+        i_gpio[i].output(0);
+
+    i_gpio[7].output(1);
+
+    configure_clock_src(clk_audio, p_mclk_in);
```

Comment:
Optional: lines 74-76 could be replaced with `spdif_tx_port_config(p_spdif_tx, clk_audio, p_mclk_in, 7)` to show off that library function.